tokio-quiche/src/quic/io/worker.rs

```diff
@@ -492,13 +591,50 @@ where
     async fn flush_buffer_to_socket(&mut self, send_buf: &[u8]) {
         if self.write_state.bytes_written > 0 {
             let current_send_buf = &send_buf[..self.write_state.bytes_written];
+
+            let socket_index = match self.write_state.network_path_id {
+                Some(path_id) => {
+                    // Always use socket 0 regardless of path ID if there is only
+                    // one (used by the server).
+                    if self.sockets.len() == 1 {
+                        0
+                    } else {
+                        path_id
+                    }
+                },
+                None => {
+                    log::error!(
+                        "{}: No network path ID available for sending data",
+                        self.id
+                    );
+                    self.metrics.write_errors(labels::QuicWriteError::Err).inc();
+                    return;
+                },
+            };
```

Comment:
...as confirmed by this first line. Also, I do not see why you could not leverage the fact that `write_socket.socket_id` is an `Option<usize>` to remove the complexity of the following lines. You should always be able to have a socket for a given address (otherwise, how could you create a packet from it?). And if such an address or a socket does not exist, it should have been detected much before.

Knowing this, I think this block could be simplified into this following line.
```suggestion
            let socket_index = self.write_state.socket_id.unwrap_or(0);
```